### PR TITLE
Roxygen colors, completion + LintR

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#1317 Colorize inside Roxygen block 
#3024 Auto Completion of Roxygen Tags 
#2767 Consider adding support for lintr 

+ tests

### CODE REVIEW

File: src/Common/Core/Impl/Services/ServiceManager.cs [hunk 1]
```diff
@@ -54,7 +54,19 @@ public class ServiceManager : IServiceManager {
         public virtual IServiceManager AddService(Type type) {
             _disposeToken.ThrowIfDisposed();
 
-            var lazy = new Lazy<object>(() => Activator.CreateInstance(type));
+            object parameter = null;
```
Comment (AlexanderSher) [inline src/Common/Core/Impl/Services/ServiceManager.cs]: There is an extension method that uses compiled expression tree: `ServiceManagerExtensions.AddService`. I'm actually not sure if we need `ServiceManager.AddService(Type type)` at all

Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Services/ServiceManager.cs]: OK, I'll see if that one is sufficient

File: src/Package/Impl/Editors/LanguageSettingsStorage.cs [hunk 2]
```diff
@@ -24,27 +24,37 @@ public sealed class LanguageSettingsStorage : IVsTextManagerEvents4, IWritableEd
...
 
         /// <summary>
+        /// Loads settings via language (editor) tools options page
+        /// </summary>
+        public void Load() {
+            LoadLanguagePreferences();
+
+            var vsShell = _services.GetService<IVsShell>(typeof(SVsShell));
+            foreach (var curAutomationObjectName in _automationObjectNames) {
+                _package.GetAutomationObject(curAutomationObjectName, out object automationObject);
```
Comment (AlexanderSher) [inline src/Package/Impl/Editors/LanguageSettingsStorage.cs]: So `automationObject` isn't used, correct?

File: src/Package/Impl/Options/R/Editor/RLintOptionsDialog.cs [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+using Microsoft.VisualStudio.R.Package.Shell;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.R.Package.Options.R.Editor {
+    public class RLintOptionsDialog: DialogPage {
+        private readonly IWritableREditorSettings _settings;
+
+        public RLintOptionsDialog() {
+            SettingsRegistryPath = @"UserSettings\R_Lint";
+            _settings = VsAppShell.Current.GetService<IWritableREditorSettings>();
```
Comment (AlexanderSher) [inline src/Package/Impl/Options/R/Editor/RLintOptionsDialog.cs]: Can we reference `VsAppShell.Current.GetService<IWritableREditorSettings>().LintOptions` directly?

File: src/Package/Impl/Editors/LanguageSettingsStorage.cs (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/Editors/LanguageSettingsStorage.cs]: Nope, this is just some acrobatics to cause VS to load settings.

File: src/Package/Impl/Options/R/Editor/RLintOptionsDialog.cs (hunk 3 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/Options/R/Editor/RLintOptionsDialog.cs]: Sure, was just simpler to debug

File: src/R/Editor/Impl/Document/REditorDocument.cs [hunk 4]
```diff
@@ -16,6 +16,7 @@ namespace Microsoft.R.Editor.Document {
     /// </summary>
     public sealed class REditorDocument : IREditorDocument {
         private readonly IServiceContainer _services;
+        private readonly TreeValidator _validator;
```
Comment (AlexanderSher) [inline src/R/Editor/Impl/Document/REditorDocument.cs]: Do you need to dispose this instance?

File: src/Windows/R/Editor/Impl/Settings/REditorSettings.cs [hunk 5]
```diff
@@ -90,5 +93,7 @@ public sealed class REditorSettings: EditorSettings, IWritableREditorSettings, I
                 return _formatOptions;
             }
         }
+
+        public LintOptions LintOptions => _lintOptions;
```
Comment (AlexanderSher) [inline src/Windows/R/Editor/Impl/Settings/REditorSettings.cs]: `public LintOptions LintOptions { get; } = new LintOptions();`

File: src/R/Editor/Impl/Document/REditorDocument.cs (hunk 4 above)
Comment (MikhailArkhipov) [inline src/R/Editor/Impl/Document/REditorDocument.cs]: It is not disposable. Used to be when validator/syntax checkers could be native

**Pull request merged**